exp/views/study.py

```diff
@@ -89,10 +93,6 @@ def get_queryset(self, *args, **kwargs):
                 # TODO optimize using subquery
                 queryset = sorted(queryset, key=lambda t: t.end_date or timezone.now(), reverse=True if '-' in sort else False)
```

Comment:
There's a way to do this with annotations, I was working on it before I got distracted. Something like this:
```
Study.objects.all().annotate(begin_date=models.Max(models.Case(models.When(logs__action='deactivate', then='logs__created_at'), default='logs__created_at', output_field=models.DateTimeField()))).values('begin_date')
```

or 

```
Study.objects.all().annotate(begin_date=models.Subquery(StudyLog.objects.filter(action='activate', study=models.OuterRef('pk')).values('created_at').annotate(begin_date=models.Max('created_at')).values('begin_date'), output_field=models.DateTimeField()))
```